lib/jekyll/csv.rb

```diff
@@ -33,7 +33,25 @@ def collection_name
       end
 
       def csv_data
-        @csv_data ||= CSV.parse(csv_string, headers: true).map(&:to_hash)
+        data = CSV.parse(csv_string, headers: true).map(&:to_hash)
+        data.delete_if do |item|
+          reject = false
+          conf['mandatory'].each do |field|
```

Comment:
I think the tests are failing when there isn't a `mandatory:` key in the config. In that case `conf['mandatory']` is `nil`, and `nil` doesn't have a `.each` method defined on it.

We can fix this by casting `nil` to an array with `.to_a`:

```suggestion
          conf['mandatory'].to_a.each do |field|
```